New extractor for "https://watch.thechosen.tv/

mrFlamel opened the pull request:
<!--
    **IMPORTANT**: PRs without the template will be CLOSED
    
    Due to the high volume of pull requests, it may be a while before your PR is reviewed.
    Please try to keep your pull request focused on a single bugfix or new feature.
    Pull requests with a vast scope and/or very large diff will take much longer to review.
    It is recommended for new contributors to stick to smaller pull requests, so you can receive much more immediate feedback as you familiarize yourself with the codebase.

    PLEASE AVOID FORCE-PUSHING after opening a PR, as it makes reviewing more difficult.
-->

### Description of your *pull request* and other information

ADD DETAILED DESCRIPTION HERE

Fixes #11246 


<details open><summary>Template</summary> <!-- OPEN is intentional -->

<!--
    # PLEASE FOLLOW THE GUIDE BELOW

    - You will be asked some questions, please read them **carefully** and answer honestly
    - Put an `x` into all the boxes `[ ]` relevant to your *pull request* (like [x])
    - Use *Preview* tab to see what your *pull request* will actually look like
-->

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check those that apply and remove the others:
- [x] I am the original author of the code in this PR, and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of the code in this PR, but it is in the public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*? Check those that apply and remove the others:
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

grqz commented on yt_dlp/extractor/watchthechosen.py:
```diff
@@ -0,0 +1,60 @@
+        'info_dict': {
+            'id': '184683594325',
+            'ext': 'mp4',
+            'title': 'Season 3 Episode 2: Two by Two',
```
Do the tests pass? Are there more fields?

mrFlamel replied:
With the latest commit, yes. Also I added "viewscount".

grqz commented on yt_dlp/extractor/watchthechosen.py:
```diff
@@ -0,0 +1,72 @@
+                }''',
+            }).encode(), headers={
+                'channelid': '12884901895',
+                'content-type': 'application/json'
```
```suggestion
                'content-type': 'application/json',
```
forgot the trailing comma...

mrFlamel commented:
Yep, now should pass everything

grqz commented on yt_dlp/extractor/watchthechosen.py:
```diff
@@ -0,0 +1,89 @@
+import json
+
+from .common import InfoExtractor
+from ..utils.traversal import traverse_obj
```
```suggestion
from ..utils import unified_strdate
from ..utils.traversal import traverse_obj
```
let's use this instead

grqz commented on yt_dlp/extractor/watchthechosen.py:
```diff
@@ -0,0 +1,89 @@
+                'channelid': '12884901895',
+                'content-type': 'application/json',
+            }), ('data', 'video'))
+
+        # Helper for formatting dates in ISO 8601
+        def formatDate(date):
+            parsed_date = datetime.datetime.fromisoformat(date)
+            return parsed_date.strftime('%Y%m%d')
```
```suggestion
```

grqz commented on yt_dlp/extractor/watchthechosen.py:
```diff
@@ -0,0 +1,89 @@
+                'title': 'title',
+                'description': 'description',
+                'thumbnail': 'thumbnail',
+                'modified_date': ('updatedAt', {formatDate}),
+                'upload_date': ('createdAt', {formatDate}),
```
```suggestion
                'modified_date': ('updatedAt', {unified_strdate}),
                'upload_date': ('createdAt', {unified_strdate}),
```

Kicer86 commented:
For S03E02 it is the same:

```
[debug] Command-line config: ['--abort-on-error', '--abort-on-unavailable-fragments', '-R', '10', '--fragment-retries', '20', '--retry-sleep', 'http:exp=1:20', '--retry-sleep', 'fragment:exp=1:60', '-N', '1', '--sleep-requests', '0.75', '--min-sleep-interval', '10', '--max-sleep-interval', '20', '-f', '(bv*[vcodec^=av01]/bv*)+mergeall[vcodec=none]', '--audio-multistreams', '--merge-output-format', 'mkv', '--sub-langs', 'all,-live_chat', '--all-subs', '--sub-format', 'best', '--embed-subs', '--embed-metadata', '--add-chapters', '--embed-thumbnail', '-v', '--check-formats', 'https://watch.thechosen.tv/video/184683594325']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2025.08.27 from yt-dlp/yt-dlp [8cd37b85d] (source)
[debug] Lazy loading extractors is disabled
[debug] Git HEAD: 067c78890
[debug] Python 3.13.7 (CPython x86_64 64bit) - Linux-6.12.46-1-lts-x86_64-with-glibc2.42 (OpenSSL 3.5.2 5 Aug 2025, glibc 2.42)
[debug] exe versions: ffmpeg 7.1.1 (setts), ffprobe 7.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.23.0, brotlicffi-1.1.0.0, certifi-2025.08.03, mutagen-1.47.0, requests-2.32.5, secretstorage-3.3.3, sqlite3-3.50.4, urllib3-2.5.0, websockets-15.0.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Plugin directories: none
[debug] Loaded 1841 extractors
[WatchTheChosen] Extracting URL: https://watch.thechosen.tv/video/184683594325
[WatchTheChosen] 184683594325: Downloading JSON metadata
[WatchTheChosen] Sleeping 0.75 seconds ...
[WatchTheChosen] 184683594325: Downloading m3u8 information
[info] 184683594325: Downloading subtitles: af, sq, am, ar, ar_eg, ar_sy, hy, bn, bn_in, bg, ceb, zh_hk, zh_hans, zh_tw, hr, cs, da, nl, en, et, fa, fi, vls_be, fr, gl, ka, de, el, gu, he, hi, hu, id, it, ja, ki, ko, ky, lv, lt, mk, ms, ml, mn, ne, no, pl, pt, pt_br, pa, ro, ru, sr, sk, sl, es, es_419, sv, tl, tg, ta, te, th, tr, uk, ur, ug, vi
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec, channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[info] Testing format 34150
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 1052
[download] Destination: /tmp/tmp8vguubdl.tmp
[download] 100% of   23.68MiB in 00:00:00 at 29.22MiB/s
[info] Testing format AUDIO-Amharic
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Amharic. Skipping...
[info] Testing format AUDIO-Arabic
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Arabic. Skipping...
[info] Testing format AUDIO-Arabic__Egypt_
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Arabic__Egypt_. Skipping...
[info] Testing format AUDIO-Arabic__Syria_
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Arabic__Syria_. Skipping...
[info] Testing format AUDIO-Bengali
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Bengali. Skipping...
[info] Testing format AUDIO-Bengali__India_
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Bengali__India_. Skipping...
[info] Testing format AUDIO-Bulgarian
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Bulgarian. Skipping...
[info] Testing format AUDIO-Cebuano
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Cebuano. Skipping...
[info] Testing format AUDIO-Chinese__Hong_Kong_
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Chinese__Hong_Kong_. Skipping...
[info] Testing format AUDIO-Chinese__Simplified_
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Chinese__Simplified_. Skipping...
[info] Testing format AUDIO-Chinese__Taiwan_
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Chinese__Taiwan_. Skipping...
[info] Testing format AUDIO-Czech
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Czech. Skipping...
[info] Testing format AUDIO-English_Audio_Descriptions
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-English_Audio_Descriptions. Skipping...
[info] Testing format AUDIO-French__France_
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-French__France_. Skipping...
[info] Testing format AUDIO-Georgian
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Georgian. Skipping...
[info] Testing format AUDIO-German
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-German. Skipping...
[info] Testing format AUDIO-Greek
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Greek. Skipping...
[info] Testing format AUDIO-Gujarati
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Gujarati. Skipping...
[info] Testing format AUDIO-Hebrew
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Hebrew. Skipping...
[info] Testing format AUDIO-Hindi
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Hindi. Skipping...
[info] Testing format AUDIO-Hungarian
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Hungarian. Skipping...
[info] Testing format AUDIO-Indonesian
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Indonesian. Skipping...
[info] Testing format AUDIO-Italian
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Italian. Skipping...
[info] Testing format AUDIO-Japanese
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Japanese. Skipping...
[info] Testing format AUDIO-Korean
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Korean. Skipping...
[info] Testing format AUDIO-Latvian
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Latvian. Skipping...
[info] Testing format AUDIO-Lithuanian
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Lithuanian. Skipping...
[info] Testing format AUDIO-Malay
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Malay. Skipping...
[info] Testing format AUDIO-Polish
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Polish. Skipping...
[info] Testing format AUDIO-Polish_lektor
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Polish_lektor. Skipping...
[info] Testing format AUDIO-Portuguese
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Portuguese. Skipping...
[info] Testing format AUDIO-Portuguese__Brazil_
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Portuguese__Brazil_. Skipping...
[info] Testing format AUDIO-Punjabi
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Punjabi. Skipping...
[info] Testing format AUDIO-Romanian
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Romanian. Skipping...
[info] Testing format AUDIO-Russian
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Russian. Skipping...
[info] Testing format AUDIO-Serbian
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Serbian. Skipping...
[info] Testing format AUDIO-Slovak
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Slovak. Skipping...
[info] Testing format AUDIO-Slovenian
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Slovenian. Skipping...
[info] Testing format AUDIO-Spanish
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Spanish. Skipping...
[info] Testing format AUDIO-Spanish__Latin_America_
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Spanish__Latin_America_. Skipping...
[info] Testing format AUDIO-Tagalog
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Tagalog. Skipping...
[info] Testing format AUDIO-Tamil
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Tamil. Skipping...
[info] Testing format AUDIO-Telugu
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Telugu. Skipping...
[info] Testing format AUDIO-Thai
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Thai. Skipping...
[info] Testing format AUDIO-Turkish
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Turkish. Skipping...
[info] Testing format AUDIO-Ukrainian
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Ukrainian. Skipping...
[info] Testing format AUDIO-Vietnamese
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-Vietnamese. Skipping...
[info] Testing format AUDIO-English
[hlsnative] Downloading m3u8 manifest
WARNING: Unable to download format AUDIO-English. Skipping...
ERROR: [WatchTheChosen] 184683594325: Requested format is not available. Use --list-formats for a list of available formats
Traceback (most recent call last):
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/YoutubeDL.py", line 1668, in wrapper
    return func(self, *args, **kwargs)
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/YoutubeDL.py", line 1824, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/YoutubeDL.py", line 1883, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/YoutubeDL.py", line 3027, in process_video_result
    raise ExtractorError(
        'Requested format is not available. Use --list-formats for a list of available formats',
        expected=True, video_id=info_dict['id'], ie=info_dict['extractor'])
yt_dlp.utils.ExtractorError: [WatchTheChosen] 184683594325: Requested format is not available. Use --list-formats for a list of available formats

```

And if I skip `--check-formats`:

```
[debug] Command-line config: ['--abort-on-error', '--abort-on-unavailable-fragments', '-R', '10', '--fragment-retries', '20', '--retry-sleep', 'http:exp=1:20', '--retry-sleep', 'fragment:exp=1:60', '-N', '1', '--sleep-requests', '0.75', '--min-sleep-interval', '10', '--max-sleep-interval', '20', '-f', '(bv*[vcodec^=av01]/bv*)+mergeall[vcodec=none]', '--audio-multistreams', '--merge-output-format', 'mkv', '--sub-langs', 'all,-live_chat', '--all-subs', '--sub-format', 'best', '--embed-subs', '--embed-metadata', '--add-chapters', '--embed-thumbnail', '-v', 'https://watch.thechosen.tv/video/184683594325']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2025.08.27 from yt-dlp/yt-dlp [8cd37b85d] (source)
[debug] Lazy loading extractors is disabled
[debug] Git HEAD: 067c78890
[debug] Python 3.13.7 (CPython x86_64 64bit) - Linux-6.12.46-1-lts-x86_64-with-glibc2.42 (OpenSSL 3.5.2 5 Aug 2025, glibc 2.42)
[debug] exe versions: ffmpeg 7.1.1 (setts), ffprobe 7.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.23.0, brotlicffi-1.1.0.0, certifi-2025.08.03, mutagen-1.47.0, requests-2.32.5, secretstorage-3.3.3, sqlite3-3.50.4, urllib3-2.5.0, websockets-15.0.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Plugin directories: none
[debug] Loaded 1841 extractors
[WatchTheChosen] Extracting URL: https://watch.thechosen.tv/video/184683594325
[WatchTheChosen] 184683594325: Downloading JSON metadata
[WatchTheChosen] Sleeping 0.75 seconds ...
[WatchTheChosen] 184683594325: Downloading m3u8 information
[info] 184683594325: Downloading subtitles: af, sq, am, ar, ar_eg, ar_sy, hy, bn, bn_in, bg, ceb, zh_hk, zh_hans, zh_tw, hr, cs, da, nl, en, et, fa, fi, vls_be, fr, gl, ka, de, el, gu, he, hi, hu, id, it, ja, ki, ko, ky, lv, lt, mk, ms, ml, mn, ne, no, pl, pt, pt_br, pa, ro, ru, sr, sk, sl, es, es_419, sv, tl, tg, ta, te, th, tr, uk, ur, ug, vi
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec, channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[info] 184683594325: Downloading 1 format(s): 34150+AUDIO-English+AUDIO-Vietnamese+AUDIO-Ukrainian+AUDIO-Turkish+AUDIO-Thai+AUDIO-Telugu+AUDIO-Tamil+AUDIO-Tagalog+AUDIO-Spanish__Latin_America_+AUDIO-Spanish+AUDIO-Slovenian+AUDIO-Slovak+AUDIO-Serbian+AUDIO-Russian+AUDIO-Romanian+AUDIO-Punjabi+AUDIO-Portuguese__Brazil_+AUDIO-Portuguese+AUDIO-Polish_lektor+AUDIO-Polish+AUDIO-Malay+AUDIO-Lithuanian+AUDIO-Latvian+AUDIO-Korean+AUDIO-Japanese+AUDIO-Italian+AUDIO-Indonesian+AUDIO-Hungarian+AUDIO-Hindi+AUDIO-Hebrew+AUDIO-Gujarati+AUDIO-Greek+AUDIO-German+AUDIO-Georgian+AUDIO-French__France_+AUDIO-English_Audio_Descriptions+AUDIO-Czech+AUDIO-Chinese__Taiwan_+AUDIO-Chinese__Simplified_+AUDIO-Chinese__Hong_Kong_+AUDIO-Cebuano+AUDIO-Bulgarian+AUDIO-Bengali__India_+AUDIO-Bengali+AUDIO-Arabic__Syria_+AUDIO-Arabic__Egypt_+AUDIO-Arabic+AUDIO-Amharic
[info] Writing video subtitles to: Season 3 Episode 2： Two by Two [184683594325].af.vtt
[debug] Invoking hlsnative downloader on "https://fastly.frontrowcdn.com/channels/12884901895/VIDEO/184683594325/279172923806.m3u8"
[hlsnative] Downloading m3u8 manifest
ERROR: Unable to download video subtitles for 'af': HTTP Error 404: Not Found
Traceback (most recent call last):
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/YoutubeDL.py", line 4428, in _write_subtitles
    self.dl(sub_filename, sub_copy, subtitle=True)
    ~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/YoutubeDL.py", line 3254, in dl
    return fd.download(name, new_info, subtitle)
           ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/downloader/common.py", line 480, in download
    ret = self.real_download(filename, info_dict)
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/downloader/hls.py", line 82, in real_download
    urlh = self.ydl.urlopen(self._prepare_url(info_dict, man_url))
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/YoutubeDL.py", line 4239, in urlopen
    return self._request_director.send(req)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/networking/common.py", line 117, in send
    response = handler.send(request)
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/networking/_helper.py", line 194, in wrapper
    return func(self, *args, **kwargs)
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/networking/common.py", line 359, in send
    return self._send(request)
           ~~~~~~~~~~^^^^^^^^^
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/networking/_requests.py", line 357, in _send
    raise HTTPError(res, redirect_loop=max_redirects_exceeded)
yt_dlp.networking.exceptions.HTTPError: HTTP Error 404: Not Found
```

As you can see I go for all streams. For a simple invocation with url only, it seems to work.

mrFlamel commented:
@Kicer86 Try now

Kicer86 commented:
@mrFlamel do you consider adding support for downloading playlists? because os of now it doesn't seem to be working

mrFlamel commented:
@Kicer86 you mean like https://watch.thechosen.tv/group/309237658599? Or https://watch.thechosen.tv/page/season%202?

Kicer86 commented:
@mrFlamel that's right

mrFlamel commented:
Added support for /group/. Fixed tests md5 checksum. Changed hls url source to prevent 404 errors.

Tried implementing /page/ but due to frontcrow api quirk, I can't know /page/ id without doing search on website and in api, so I think it is too much of pain. If anybody wants to add that functionality then feel free to.

@Kicer86 you can test ;)

Kicer86 commented:
@mrFlamel fetching 'groups' looks ok (at least those I tried). But I get 404 on first season's specials

mrFlamel commented:
@Kicer86 I couldn't reproduce again... -v please?

Kicer86 commented:
@mrFlamel 
```
[debug] Command-line config: ['--abort-on-error', '--abort-on-unavailable-fragments', '-R', '10', '--fragment-retries', '20', '--retry-sleep', 'http:exp=1:20', '--retry-sleep', 'fragment:exp=1:60', '-N', '1', '--sleep-requests', '0.75', '--min-sleep-interval', '10', '--max-sleep-interval', '20', '--check-formats', '-f', '(bv*[vcodec^=av01]/bv*)+mergeall[vcodec=none]', '--audio-multistreams', '--merge-output-format', 'mkv', '--sub-langs', 'all,-live_chat', '--all-subs', '--sub-format', 'best', '--embed-subs', '--embed-metadata', '--add-chapters', '--embed-thumbnail', '-v', 'https://watch.thechosen.tv/video/184683594365']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2025.08.27 from yt-dlp/yt-dlp [8cd37b85d] (source)
[debug] Lazy loading extractors is disabled
[debug] Git HEAD: 60d570e56
[debug] Python 3.13.7 (CPython x86_64 64bit) - Linux-6.12.47-1-lts-x86_64-with-glibc2.42 (OpenSSL 3.5.2 5 Aug 2025, glibc 2.42)
[debug] exe versions: ffmpeg 7.1.1 (setts), ffprobe 7.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.23.0, brotlicffi-1.1.0.0, certifi-2025.08.03, mutagen-1.47.0, requests-2.32.5, secretstorage-3.3.3, sqlite3-3.50.4, urllib3-2.5.0, websockets-15.0.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Plugin directories: none
[debug] Loaded 1841 extractors
[WatchTheChosen] Extracting URL: https://watch.thechosen.tv/video/184683594365
[WatchTheChosen] 184683594365: Downloading JSON metadata
[WatchTheChosen] Sleeping 0.75 seconds ...
[WatchTheChosen] 184683594365: Downloading m3u8 information
[info] 184683594365: Downloading subtitles: af, sq, ar, hy, bg, zh_hk, zh_cn, zh_tw, hr, cs, da, nl, en, et, fi, fr, de, el, hu, id, it, ja, ko, ms, ml, mn, no, pl, pt, pt_br, ro, sr, sk, es, es_419, sv, tl, th, uk, ug, vi
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec, channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[info] Testing format 16544
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 316
[download] Destination: /tmp/tmpx8saolni.tmp
[download] 100% of    9.02MiB in 00:00:00 at 20.03MiB/s
[info] Testing format AUDIO-Chinese__Hong_Kong_
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmpgpcj6rim.tmp
[download] 100% of   62.85KiB in 00:00:00 at 1.11MiB/s
[info] Testing format AUDIO-Chinese__Simplified__China_
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmpkoidbr6d.tmp
[download] 100% of   62.85KiB in 00:00:00 at 1.09MiB/s
[info] Testing format AUDIO-Chinese__Taiwan_
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmpkoaxo5tt.tmp
[download] 100% of   62.85KiB in 00:00:00 at 1.11MiB/s
[info] Testing format AUDIO-Czech
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmprbrtpwnk.tmp
[download] 100% of   62.86KiB in 00:00:00 at 1.13MiB/s
[info] Testing format AUDIO-English_Audio_Descriptions
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 316
[download] Destination: /tmp/tmpmf4a0rsg.tmp
[download] 100% of   62.84KiB in 00:00:00 at 1.10MiB/s
[info] Testing format AUDIO-French__France_
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmpqqtg78au.tmp
[download] 100% of   62.85KiB in 00:00:00 at 1.27MiB/s
[info] Testing format AUDIO-German
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmpp976q2kq.tmp
[download] 100% of   62.85KiB in 00:00:00 at 1.08MiB/s
[info] Testing format AUDIO-Greek
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmpvfne6otj.tmp
[download] 100% of   62.85KiB in 00:00:00 at 1.54MiB/s
[info] Testing format AUDIO-Hungarian
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmp7wrpcp8h.tmp
[download] 100% of   62.86KiB in 00:00:00 at 1.51MiB/s
[info] Testing format AUDIO-Indonesian
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmp_sm5fxz4.tmp
[download] 100% of   62.85KiB in 00:00:00 at 1.61MiB/s
[info] Testing format AUDIO-Italian
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmpkfntjz0k.tmp
[download] 100% of   62.85KiB in 00:00:00 at 1.56MiB/s
[info] Testing format AUDIO-Japanese
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmp6xsdmdjr.tmp
[download] 100% of   62.86KiB in 00:00:00 at 1.61MiB/s
[info] Testing format AUDIO-Korean
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmpjdaznadk.tmp
[download] 100% of   62.63KiB in 00:00:00 at 1.44MiB/s
[info] Testing format AUDIO-Malay
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmpz_25yij_.tmp
[download] 100% of   62.76KiB in 00:00:00 at 1.14MiB/s
[info] Testing format AUDIO-Polish
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 320
[download] Destination: /tmp/tmp6ucmve67.tmp
[download] 100% of   62.87KiB in 00:00:00 at 1.14MiB/s
[info] Testing format AUDIO-Polish_Lektor
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmprltij4jl.tmp
[download] 100% of   62.84KiB in 00:00:00 at 1.13MiB/s
[info] Testing format AUDIO-Portuguese
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmps3c415xz.tmp
[download] 100% of   63.02KiB in 00:00:00 at 1.09MiB/s
[info] Testing format AUDIO-Portuguese__Brazil_
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmpzvhh2j1q.tmp
[download] 100% of   62.83KiB in 00:00:00 at 1.06MiB/s
[info] Testing format AUDIO-Romanian
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 320
[download] Destination: /tmp/tmp42008ses.tmp
[download] 100% of   62.87KiB in 00:00:00 at 1.19MiB/s
[info] Testing format AUDIO-Russian
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 320
[download] Destination: /tmp/tmpmppywvqf.tmp
[download] 100% of   62.87KiB in 00:00:00 at 1.10MiB/s
[info] Testing format AUDIO-Serbian
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 320
[download] Destination: /tmp/tmp6414kba0.tmp
[download] 100% of   62.87KiB in 00:00:00 at 1.10MiB/s
[info] Testing format AUDIO-Slovak
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmpd18aou4x.tmp
[download] 100% of   62.85KiB in 00:00:00 at 1.15MiB/s
[info] Testing format AUDIO-Slovenian
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 320
[download] Destination: /tmp/tmp2wpwli4m.tmp
[download] 100% of   62.87KiB in 00:00:00 at 1.12MiB/s
[info] Testing format AUDIO-Spanish
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 320
[download] Destination: /tmp/tmpys_vvpvj.tmp
[download] 100% of   62.87KiB in 00:00:00 at 1.09MiB/s
[info] Testing format AUDIO-Spanish__Latin_America_
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmpw1qchtx0.tmp
[download] 100% of   62.85KiB in 00:00:00 at 1.16MiB/s
[info] Testing format AUDIO-Tagalog
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmpftaytpph.tmp
[download] 100% of   62.86KiB in 00:00:00 at 1.08MiB/s
[info] Testing format AUDIO-Thai
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmp_wzqdkid.tmp
[download] 100% of   62.85KiB in 00:00:00 at 1.11MiB/s
[info] Testing format AUDIO-Turkish
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmph0zm9eze.tmp
[download] 100% of   62.83KiB in 00:00:00 at 435.27KiB/s
[info] Testing format AUDIO-Ukrainian
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 317
[download] Destination: /tmp/tmpzmlfo6ra.tmp
[download] 100% of   62.85KiB in 00:00:00 at 1.13MiB/s
[info] Testing format AUDIO-English
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 316
[download] Destination: /tmp/tmpjhm00yi6.tmp
[download] 100% of   62.84KiB in 00:00:00 at 1.10MiB/s
[info] 184683594365: Downloading 1 format(s): 16544+AUDIO-English+AUDIO-Ukrainian+AUDIO-Turkish+AUDIO-Thai+AUDIO-Tagalog+AUDIO-Spanish__Latin_America_+AUDIO-Spanish+AUDIO-Slovenian+AUDIO-Slovak+AUDIO-Serbian+AUDIO-Russian+AUDIO-Romanian+AUDIO-Portuguese__Brazil_+AUDIO-Portuguese+AUDIO-Polish_Lektor+AUDIO-Polish+AUDIO-Malay+AUDIO-Korean+AUDIO-Japanese+AUDIO-Italian+AUDIO-Indonesian+AUDIO-Hungarian+AUDIO-Greek+AUDIO-German+AUDIO-French__France_+AUDIO-English_Audio_Descriptions+AUDIO-Czech+AUDIO-Chinese__Taiwan_+AUDIO-Chinese__Simplified__China_+AUDIO-Chinese__Hong_Kong_
[info] Writing video subtitles to: The Shepherd： A Story of the First Christmas [184683594365].af.vtt
[debug] Invoking hlsnative downloader on "https://fastly.frontrowcdn.com/channels/12884901895/VIDEO/184683594365/279172876613.m3u8"
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 8
[download] Destination: The Shepherd： A Story of the First Christmas [184683594365].af.vtt
[download] Got error: HTTP Error 404: Not Found. Retrying fragment 1 (1/20)...
Sleeping 1.00 seconds ...
[download] Got error: HTTP Error 404: Not Found. Retrying fragment 1 (2/20)...
Sleeping 2.00 seconds ...
[download] Got error: HTTP Error 404: Not Found. Retrying fragment 1 (3/20)...
Sleeping 4.00 seconds ...
[download] Got error: HTTP Error 404: Not Found. Retrying fragment 1 (4/20)...
Sleeping 8.00 seconds ...
[download] Got error: HTTP Error 404: Not Found. Retrying fragment 1 (5/20)...
Sleeping 16.00 seconds ...
[download] Got error: HTTP Error 404: Not Found. Retrying fragment 1 (6/20)...
Sleeping 32.00 seconds ...
^C
ERROR: Interrupted by user
Exception ignored in: <_io.FileIO name=4 mode='wb' closefd=True>
ResourceWarning: unclosed file <_io.BufferedWriter name=4>
```

mrFlamel commented:
@Kicer86 seems like a problem with yt-dlp downloader. Good vtt m3u8 files contain this: 
```
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-ALLOW-CACHE:YES
#EXT-X-TARGETDURATION:1144
#EXTINF:1144,
subtitle/279172899884/seg_0.vtt
#EXT-X-ENDLIST
```

but bad ones have this:

```
WEBVTT
X-TIMESTAMP-MAP=MPEGTS:900000,LOCAL:00:00:00.000

1
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-TARGETDURATION:301
#EXT-X-MEDIA-SEQUENCE:1
#EXT-X-PLAYLIST-TYPE:VOD
#EXTINF:300,
special-the-shepherd-s01e09-en-1080p-tc-subs-af-encoder_00001.vtt
#EXTINF:301,
special-the-shepherd-s01e09-en-1080p-tc-subs-af-encoder_00002.vtt
#EXTINF:300,
special-the-shepherd-s01e09-en-1080p-tc-subs-af-encoder_00003.vtt
#EXTINF:300,
special-the-shepherd-s01e09-en-1080p-tc-subs-af-encoder_00004.vtt
#EXTINF:64,
special-the-shepherd-s01e09-en-1080p-tc-subs-af-encoder_00005.vtt
#EXT-X-ENDLIST
```

and I couldn't download the bad ones in any way (tested yt-dlp, youtube-dl and ffmpeg). They just don't recognise m3u8 header. Idk if this file should download, but seems like yt-dlp can't download that, though watch.thechosen.tv uses that for subtitles. If yt-dlp should download that, then we have new issue

Kicer86 commented:
There is a new season 5 available and they must have changed something. I cannot download neither the whole group, nor single episode.

```
[debug] Command-line config: ['-v', 'https://watch.thechosen.tv/group/309237661102']
[debug] User config "/home/michal/.config/yt-dlp/config": ['--abort-on-error', '--abort-on-unavailable-fragments', '-R', '10', '--fragment-retries', '20', '--retry-sleep', 'http:exp=1:20', '--retry-sleep', 'fragment:exp=1:60', '--sleep-requests', '0.75', '--min-sleep-interval', '10', '--max-sleep-interval', '20', '--check-formats', '-f', '(bv*[vcodec^=av01]/bv*)+((mergeall[vcodec=none][protocol^=m3u8])/(bestaudio))', '--audio-multistreams', '--merge-output-format', 'mkv', '--sub-langs', 'all,-live_chat', '--all-subs', '--sub-format', 'best', '--embed-subs', '--embed-metadata', '--add-chapters', '--embed-thumbnail']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2025.08.27 from yt-dlp/yt-dlp [8cd37b85d] (source)
[debug] Lazy loading extractors is disabled
[debug] Git HEAD: 4918b0078
[debug] Python 3.13.7 (CPython x86_64 64bit) - Linux-6.12.48-1-lts-x86_64-with-glibc2.42 (OpenSSL 3.5.3 16 Sep 2025, glibc 2.42)
[debug] exe versions: ffmpeg 7.1.1 (setts), ffprobe 7.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.23.0, brotlicffi-1.1.0.0, certifi-2025.08.03, mutagen-1.47.0, requests-2.32.5, secretstorage-3.3.3, sqlite3-3.50.4, urllib3-2.5.0, websockets-15.0.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Plugin directories: none
[debug] Loaded 1841 extractors
[WatchTheChosen] Extracting URL: https://watch.thechosen.tv/group/309237661102
[WatchTheChosen] 309237661102: Downloading playlist metadata
ERROR: [WatchTheChosen] 184683621777: Failed to obtain m3u8 URL; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/extractor/common.py", line 762, in extract
    ie_result = self._real_extract(url)
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/extractor/watchthechosen.py", line 110, in _real_extract
    formats, subtitles = self._extract_m3u8_formats_and_subtitles(video_metadata['url'], video_metadata['id'])
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/extractor/common.py", line 2183, in _extract_m3u8_formats_and_subtitles
    raise ExtractorError(errnote, video_id=video_id)
```


```
[debug] Command-line config: ['-v', 'https://watch.thechosen.tv/video/184683621777']
[debug] User config "/home/michal/.config/yt-dlp/config": ['--abort-on-error', '--abort-on-unavailable-fragments', '-R', '10', '--fragment-retries', '20', '--retry-sleep', 'http:exp=1:20', '--retry-sleep', 'fragment:exp=1:60', '--sleep-requests', '0.75', '--min-sleep-interval', '10', '--max-sleep-interval', '20', '--check-formats', '-f', '(bv*[vcodec^=av01]/bv*)+((mergeall[vcodec=none][protocol^=m3u8])/(bestaudio))', '--audio-multistreams', '--merge-output-format', 'mkv', '--sub-langs', 'all,-live_chat', '--all-subs', '--sub-format', 'best', '--embed-subs', '--embed-metadata', '--add-chapters', '--embed-thumbnail']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2025.08.27 from yt-dlp/yt-dlp [8cd37b85d] (source)
[debug] Lazy loading extractors is disabled
[debug] Git HEAD: 4918b0078
[debug] Python 3.13.7 (CPython x86_64 64bit) - Linux-6.12.48-1-lts-x86_64-with-glibc2.42 (OpenSSL 3.5.3 16 Sep 2025, glibc 2.42)
[debug] exe versions: ffmpeg 7.1.1 (setts), ffprobe 7.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.23.0, brotlicffi-1.1.0.0, certifi-2025.08.03, mutagen-1.47.0, requests-2.32.5, secretstorage-3.3.3, sqlite3-3.50.4, urllib3-2.5.0, websockets-15.0.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Plugin directories: none
[debug] Loaded 1841 extractors
[WatchTheChosen] Extracting URL: https://watch.thechosen.tv/video/184683621777
[WatchTheChosen] 184683621777: Downloading JSON metadata
ERROR: [WatchTheChosen] 184683621777: Failed to obtain m3u8 URL; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/extractor/common.py", line 762, in extract
    ie_result = self._real_extract(url)
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/extractor/watchthechosen.py", line 147, in _real_extract
    formats, subtitles = self._extract_m3u8_formats_and_subtitles(metadata['url'], pageID)
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/michal/projects/sources/yt-dlp/yt_dlp/extractor/common.py", line 2183, in _extract_m3u8_formats_and_subtitles
    raise ExtractorError(errnote, video_id=video_id)
```

seproDev commented on yt_dlp/extractor/watchthechosen.py:
```diff
@@ -0,0 +1,185 @@
+            'comment_count': int,
+            'view_count': int,
+        },
+    }]
```
There should also be a test for a group URL

seproDev commented on yt_dlp/extractor/watchthechosen.py:
```diff
@@ -0,0 +1,185 @@
+
+                                fragment PageContainerVideo on Video {
+                                  audiences { id name __typename }
+                                  title description updatedAt thumbnail createdAt duration likeCount comments views url hasAccess id __typename
```
Do we really need all these fragments?
Maybe something like this:
```graphql
query PaginatedStaticPageContainer($channelID: ID!, $pageContainerID: ID!) {
  pageContainer(ChannelID: $channelID, PageContainerID: $pageContainerID) {
    ... on StaticPageContainer {
      id
      title
      itemRefs {
        edges {
          node {
            id
            contentItem {
              ... on ItemVideo {
                videoItem: item {
                  createdAt
                  description
                  duration
                  hasAccess
                  id
                  thumbnail
                  title
                  url
                  views
                  visibility
                }
              }
            }
          }
        }
      }
      createdAt
      updatedAt
    }
  }
}
```

seproDev commented on yt_dlp/extractor/watchthechosen.py:
```diff
@@ -0,0 +1,185 @@
+from ..utils.traversal import traverse_obj
+
+
+class WatchTheChosenIE(InfoExtractor):
```
There are other sites that make use of [Frontro](https://frontro.com/) like https://app.johnmalecki.com/video/184683636857
With that, this should probably be a generic `FrontroBaseIE` extractor that then has concrete classes for TheChosenIE etc.

mrFlamel replied:
@seproDev do you suggest completing first this extractor with these improvements you mentioned or scraping it and making new pull request FrontroBaseIE extractor, that would include different classes for different sites? As I would imagine, the second option is a lot more effort for lot less result, since The Chosen is certainly the biggest in Frontro sites and other success stories have never had any issues opened or closed, (as for opposed to The Chosen, who had 3 issues opened (2 as duplicate) since 2024 migration from angelstudios.com).

seproDev replied:
Nothing needs to be scrapped. Since the API is shared among all the sites, you can just move your existing code in to an abstract base class (i.e. `FrontroBaseIE`) and then inherit from that with `TheChosenIE`. So something like this:
```py
class FrontroBaseIE(InfoExtractor):
   ... # Your existing video extraction code

class FrontroGroupBaseIE(InfoExtractor):
   ... # Your existing Group extraction code

class TheChosenIE(FrontroBaseIE):
   ... # Inherits real_extract and only sets different VALID_URL and channelid

class TheChosenGroupIE(FrontroGroupBaseIE):
   ... # Inherits real_extract and only sets different VALID_URL and channelid
```

mrFlamel commented:
@seproDev thank you for being patient with these code changes. I restructured the whole thing now for Frontro abstract class and changed errors to more generic. I would imagine that adding more sites (like app.johnmalecki.com) would require separate PR though or?

@Kicer86 Am I correct that this is Season 1 Episode 1 and "ca" subtitles? Also, it is probably frontro and their (possibly ai generated) subtitle files problem. Maybe yt-dlp postprocessor could fix such corruptions if they happen? (Idea for new issue)

mrFlamel commented:
@Kicer86 running `./yt-dlp.sh https://watch.thechosen.tv/video/184683594334 --list-subs` doesn't show any 'cat' language?

Kicer86 commented:
> running `./yt-dlp.sh https://watch.thechosen.tv/video/184683594334 --list-subs` doesn't show any 'cat' language?

https://en.wikipedia.org/wiki/List_of_ISO_639_language_codes tells that cat == ca in 2 letter codes.

mrFlamel commented:
Idk maybe I'm doing something wrong, but this "ca" language looks absolutely fine. If you can work on that problem though, then I'd appreciate it

mrFlamel commented:
@Kicer86 I think the video has been updated. Frontro query says "updatedAt": "2025-10-31T13:25:59.627831Z" so they probably just added new languages. These unresolved language codes do appear in ISO 693-3 (so 2007 revision), which is also listed here: https://en.wikipedia.org/wiki/List_of_ISO_639-3_codes. They are probably smaller languages, that vlc doesn't resolve (maybe these languages don't have names in different languages?).
To be honest (and I may be wrong, but), I think these weird anomalies like 'und' and 'afr' with zero length are probably human-made mistakes by these people who upload and translate The Chosen movies. As watch.thechosen.tv doesn't offer official download methods, then these people aren't probably bothered by corrupt files, that work in their system.

mrFlamel commented:
@seproDev (why doesn't it offer ping) my only question is why modified_date and modified_timestamp are as a date, when they change often (like I run tests today and they had already been changed)

mrFlamel commented:
# 🥳
